--- origsrc/fltk-1.1.9/configure.in	2008-04-11 11:53:01.000000000 -0500
+++ src/fltk-1.1.9/configure.in	2008-06-30 14:04:53.843750000 -0500
@@ -52,7 +52,7 @@
 fi
 if test "x$uname" = x; then
     # MingW doesn't provide any output when uname is run, even with "-s"...
-    uname="CYGWIN"
+    uname="MINGW"
 fi
 
 dnl Don't automatically add "-g" to compiler options...
@@ -69,7 +69,7 @@
 
 dnl OS-specific pre-tests...
 case $uname in
-    CYGWIN* | MINGW*)
+    MINGW*)
         # Handle Cygwin option *first*, before all other tests.
 	AC_ARG_ENABLE(cygwin, [  --enable-cygwin         use the CygWin libraries [default=no]])
 	if test x$enable_cygwin != xyes; then
@@ -192,7 +192,7 @@
             DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -Wl,-bexpall,-bM:SRE,-bnoentry -o"
             SHAREDSUFFIX="_s"
             ;;
-        CYGWIN* | MINGW*)
+        MINGW*)
 	    PICFLAG=0
 	    if test x$enable_cygwin != xyes; then
 		DSONAME="mgwfltknox-$FL_API_VERSION.dll"
@@ -215,6 +215,18 @@
 		-Wl,--enable-runtime-pseudo-reloc -Wl,--enable-auto-import \
 		-Wl,--enable-auto-image-base -o \$@"
 	    ;;
+	CYGWIN*)
+	    PICFLAG=0
+		DSONAME="cygfltk-$FL_API_VERSION.dll"
+		FLDSONAME="cygfltk_forms-$FL_API_VERSION.dll"
+		GLDSONAME="cygfltk_gl-$FL_API_VERSION.dll"
+		IMGDSONAME="cygfltk_images-$FL_API_VERSION.dll"
+		LDFLAGS="$LDFLAGS -Wl,--enable-auto-import -Wl,--enable-runtime-pseudo-reloc"
+	    DSOCOMMAND="\$(CXX) \$(DSOFLAGS) -shared \
+	    	-Wl,--whole-archive -Wl,--export-all-symbols \
+		-Wl,--enable-runtime-pseudo-reloc -Wl,--enable-auto-import \
+		-Wl,--enable-auto-image-base -o \$@"
+		;;
 	*)
             AC_MSG_WARN(Shared libraries may not be supported.  Trying -shared option with compiler.)
             DSONAME="libfltk.so.$FL_API_VERSION"
@@ -460,7 +472,7 @@
 AUDIOLIBS=""
 
 case $uname in
-    CYGWIN* | MINGW*)
+    MINGW*)
 	dnl Cygwin environment...
 	AUDIOLIBS="-lwinmm"
 	;;
@@ -609,7 +621,7 @@
 UNINSTALL_DESKTOP=""
 
 case $uname in
-    CYGWIN* | MINGW*)
+    MINGW*)
 	dnl Cygwin environment...
 	# Recent versions of Cygwin are seriously broken and the size
 	# checks don't work because the shell puts out \r\n instead of
@@ -693,7 +705,7 @@
 	# All others are UNIX/X11...
 	if test x$have_pthread = xyes; then
 	    AC_DEFINE(HAVE_PTHREAD)
-	    THREADS="threads"
+	    THREADS="threads$EXEEXT"
 	fi
 
 	dnl Check for X11...
@@ -707,10 +719,9 @@
 	    AC_MSG_WARN(Ignoring libraries \"$X_PRE_LIBS\" requested by configure.)
 	fi
 
-	LIBS="$LIBS -lXext -lX11 $X_EXTRA_LIBS"
+	LIBS="$LIBS -lXext -lX11 $X_EXTRA_LIBS $X_LIBS"
 	CFLAGS="$CFLAGS $X_CFLAGS"
 	CXXFLAGS="$CXXFLAGS $X_CFLAGS"
-	LDFLAGS="$X_LIBS $LDFLAGS"
 	DSOFLAGS="$X_LIBS $DSOFLAGS"
 
 	if test "x$x_includes" != x; then
@@ -1080,7 +1091,7 @@
 echo "-------------------------------------------------------------------------"
 
 case $uname in
-    CYGWIN* | MINGW*)
+    MINGW*)
         graphics="GDI"
 	;;
     Darwin*)
--- origsrc/fltk-1.1.9/src/Makefile	2008-02-27 09:00:31.000000000 -0600
+++ src/fltk-1.1.9/src/Makefile	2008-06-30 14:04:53.859375000 -0500
@@ -354,24 +354,24 @@
 #   against the import libraries in the src dir.
 #-----------------------------------------------------------------
 
-cygfltknox-1.1.dll: $(LIBNAME)
+cygfltk-1.1.dll: $(LIBNAME)
 	echo $(DSOCOMMAND) $(LIBNAME) ...
 	$(DSOCOMMAND) $(LIBNAME) -Wl,--no-whole-archive \
 		-Wl,--out-implib=libfltk.dll.a $(LDLIBS)
 
-cygfltknox_forms-1.1.dll: $(FLLIBNAME) cygfltknox-1.1.dll
+cygfltk_forms-1.1.dll: $(FLLIBNAME) cygfltk-1.1.dll
 	echo $(DSOCOMMAND) $(FLLIBNAME) ...
 	$(DSOCOMMAND) $(FLLIBNAME) -Wl,--no-whole-archive \
 		-Wl,--out-implib=libfltk_forms.dll.a \
 		-L. -lfltk $(LDLIBS)
 
-cygfltknox_gl-1.1.dll: $(GLLIBNAME) cygfltknox-1.1.dll
+cygfltk_gl-1.1.dll: $(GLLIBNAME) cygfltk-1.1.dll
 	echo $(DSOCOMMAND) $(GLLIBNAME) ...
 	$(DSOCOMMAND) $(GLLIBNAME) -Wl,--no-whole-archive \
 		-Wl,--out-implib=libfltk_gl.dll.a \
 		-L. -lfltk $(GLDLIBS)
 
-cygfltknox_images-1.1.dll: $(IMGLIBNAME) cygfltknox-1.1.dll
+cygfltk_images-1.1.dll: $(IMGLIBNAME) cygfltk-1.1.dll
 	echo $(DSOCOMMAND) $(IMGLIBNAME) ...
 	$(DSOCOMMAND) $(IMGLIBNAME) -Wl,--no-whole-archive \
 		-Wl,--out-implib=libfltk_images.dll.a \
@@ -523,7 +523,7 @@
 		$(RM) $(DESTDIR)$(libdir)/libfltk_s.a;\
 		$(INSTALL_LIB) libfltk_s.a $(DESTDIR)$(libdir); \
 	fi
-	if test x$(DSONAME) = xcygfltknox-1.1.dll; then\
+	if test x$(DSONAME) = xcygfltk-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(DSONAME);\
 		$(INSTALL_LIB) $(DSONAME) $(DESTDIR)$(bindir); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk.dll.a;\
@@ -554,7 +554,7 @@
 		$(RM) $(DESTDIR)$(libdir)/libfltk_forms_s.a;\
 		$(INSTALL_LIB) libfltk_forms_s.a $(DESTDIR)$(libdir); \
 	fi
-	if test x$(FLDSONAME) = xcygfltknox_forms-1.1.dll; then\
+	if test x$(FLDSONAME) = xcygfltk_forms-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(FLDSONAME);\
 		$(INSTALL_LIB) $(FLDSONAME) $(DESTDIR)$(bindir); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk_forms.dll.a;\
@@ -585,7 +585,7 @@
 		$(RM) $(DESTDIR)$(libdir)/libfltk_gl_s.a;\
 		$(INSTALL_LIB) libfltk_gl_s.a $(DESTDIR)$(libdir); \
 	fi
-	if test x$(GLDSONAME) = xcygfltknox_gl-1.1.dll; then\
+	if test x$(GLDSONAME) = xcygfltk_gl-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(GLDSONAME);\
 		$(INSTALL_LIB) $(GLDSONAME) $(DESTDIR)$(bindir); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk_gl.dll.a;\
@@ -616,7 +616,7 @@
 		$(RM) $(DESTDIR)$(libdir)/libfltk_images_s.a;\
 		$(INSTALL_LIB) libfltk_images_s.a $(DESTDIR)$(libdir); \
 	fi
-	if test x$(IMGDSONAME) = xcygfltknox_images-1.1.dll; then\
+	if test x$(IMGDSONAME) = xcygfltk_images-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(IMGDSONAME); \
 		$(INSTALL_LIB) $(IMGDSONAME) $(DESTDIR)$(bindir); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk_images.dll.a;\
@@ -644,7 +644,7 @@
 	if test x$(DSONAME) = xlibfltk_s.a; then\
 		$(RM) $(DESTDIR)$(libdir)/libfltk_s.a;\
 	fi
-	if test x$(DSONAME) = xcygfltknox-1.1.dll; then\
+	if test x$(DSONAME) = xcygfltk-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(DSONAME); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk.dll.a;\
 	fi
@@ -665,7 +665,7 @@
 	if test x$(FLDSONAME) = xlibfltk_forms_s.a; then\
 		$(RM) $(DESTDIR)$(libdir)/libfltk_forms_s.a;\
 	fi
-	if test x$(FLDSONAME) = xcygfltknox_forms-1.1.dll; then\
+	if test x$(FLDSONAME) = xcygfltk_forms-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(FLDSONAME); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk_forms.dll.a;\
 	fi
@@ -688,7 +688,7 @@
 	if test x$(GLDSONAME) = xlibfltk_gl_s.a; then\
 		$(RM) $(DESTDIR)$(libdir)/libfltk_gl_s.a;\
 	fi
-	if test x$(GLDSONAME) = xcygfltknox_gl-1.1.dll; then\
+	if test x$(GLDSONAME) = xcygfltk_gl-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(GLDSONAME); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk_gl.dll.a;\
 	fi
@@ -711,7 +711,7 @@
 	if test x$(IMGDSONAME) = xlibfltk_images_s.a; then\
 		$(RM) $(DESTDIR)$(libdir)/libfltk_images_s.a;\
 	fi
-	if test x$(IMGDSONAME) = xcygfltknox_images-1.1.dll; then\
+	if test x$(IMGDSONAME) = xcygfltk_images-1.1.dll; then\
 		$(RM) $(DESTDIR)$(bindir)/$(IMGDSONAME); \
 		$(RM) $(DESTDIR)$(libdir)/libfltk_images.dll.a;\
 	fi
